lib/db/schemaupdater.go

```diff
@@ -405,13 +401,9 @@ func (db *schemaUpdater) updateSchema6to7(_ int) error {
 				}
 				delErr = t.Delete(key)
 				return delErr == nil
-			}
-			var fl VersionList
-			err = fl.Unmarshal(svl)
-			if err != nil {
-				// This can't happen, but it's ignored everywhere else too,
-				// so lets not act on it.
-				return true
+			} else if err != nil {
+				delErr = err
+				return false
```

Comment:
Again, different behaviour in case of marshal error?

Reply:
change undone